### PR TITLE
fix(analytics-js-integrations): resolved tech debt items

### PR DESCRIPTION
**Pull request by mihir-4116**
## PR Description

- This pull request addresses below changes

1. Removed exceptions thrown to core sdk from integrations
2. Removed reference of analytics object from event level 
3. Resolved eslint errors for integrations

## Linear task (optional)

Linear task link

## Cross Browser Tests

Please confirm you have tested for the following browsers:

- [ ] Chrome
- [ ] Firefox
- [ ] IE11

## Sanity Suite

- [ ] All sanity suite test cases pass locally

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: packages/analytics-js-integrations/src/integrations/DCMFloodlight/utils.js [hunk 1]
```diff
@@ -99,7 +99,7 @@ const mapFlagValue = (key, value) => {
     return 0;
   }
 
-  throw Error(`[DCM Floodlight]:: ${key}: valid parameters are [1|true] or [0|false]`);
+  return 0;
```
Comment (shrouti1507) [inline packages/analytics-js-integrations/src/integrations/DCMFloodlight/utils.js:102]: does this mean we are not offering any visibility to the user about why the event is dropped?

File: packages/analytics-js-integrations/src/integrations/PinterestTag/browser.js [hunk 2]
```diff
@@ -230,8 +230,9 @@ export default class PinterestTag {
     window.pintrk('track', event, pageObject);
   }
 
-  identify() {
-    const userTraits = this.analytics.getUserTraits();
+  identify(rudderElement) {
+    const { context } = rudderElement.message;
+    const userTraits = context?.traits || {};
```
Comment (ItsSudip) [inline packages/analytics-js-integrations/src/integrations/PinterestTag/browser.js:235]: Don't we need to consider traits.email?

File: packages/analytics-js-integrations/src/integrations/DCMFloodlight/utils.js (hunk 1 above)
Comment (mihir-4116) [reply packages/analytics-js-integrations/src/integrations/DCMFloodlight/utils.js:102]: we can use logger.error to show error message but we can't throw an exception to core SDK. in this case we can always fall back to default value (which is 0) so there is no need to throw an error

File: packages/analytics-js-integrations/src/integrations/PinterestTag/browser.js (hunk 2 above)
Comment (mihir-4116) [reply packages/analytics-js-integrations/src/integrations/PinterestTag/browser.js:235]: no in js-sdk user traits or identify call should be referred from context.traits only and for group call we should refer from root level traits. no need to take it from traits.email

**Review by ItsSudip — APPROVED** [review]

**Review by shrouti1507 — APPROVED** [review]

**Review by koladilip — APPROVED** [review]

**Pull request merged**